apps/aefate/src/aefa_engine_state.erl

```diff
@@ -100,9 +122,22 @@
 -include_lib("aecontract/include/aecontract.hrl").
 -include_lib("aecontract/include/hard_forks.hrl").
 
+-type debugger_status() :: break | next | step | continue.
+-type debugger_location() :: none | {string(), integer()}.
+
 -type void_or_fate() :: ?FATE_VOID | aeb_fate_data:fate_type().
 -type pubkey() :: <<_:256>>.
 
+-record(debug_info, { status = continue     :: debugger_status()
+                    , location = none       :: debugger_location()
+                    , breakpoints = []      :: list()
+                    , skip_instructions = 0 :: integer()
+                    , vars_registers = #{}  :: map()
+                    , call_stack = []       :: [{string(), pos_integer()}]
```

Comment:
Maybe also keep contract call stack that would remember contract names and addresses?

Reply:
Solved here by adding `DBG_CONTRACT` https://github.com/aeternity/aeternity/pull/4079/commits/b8213daa5f6d63d0cc312e8a5da11b88f2b07b12.